sympy/solvers/bivariate.py

```diff
@@ -205,11 +251,36 @@ def _solve_lambert(f, symbol, gens):
     if not lamcheck:
         raise NotImplementedError()
 
-    if lhs.is_Mul:
-        lhs = expand_log(log(lhs))
-        rhs = log(rhs)
+    if lhs.is_Add or lhs.is_Mul:
+        # replacing all even_degrees of symbol with dummy variable t
+        # since these will need special handling; non-Add/Mul do not
+        # need this handling
+        t = Dummy('t', **symbol.assumptions0)
+        lhs = lhs.replace(
+            lambda i:  # find symbol**even
+                i.is_Pow and i.base == symbol and i.exp % 2 == 0,
```

Comment:
```suggestion
                i.is_Pow and i.base == symbol and i.exp.is_even,
```

Reply:
This may make it more transparent if someone is not accustomed to reading `x % 2 == 0' as 'even'. Note, too, that either form will enable solutions to `x**(2*var('i', integer=True))*exp(x/2) - 1`.

Reply:
Done